pgrx-sql-entity-graph/src/pg_extern/returning.rs

```diff
@@ -111,62 +102,40 @@ impl Returning {
-                                }
-                                _ => (),
+
+                    loop {
+                        if let Some(segment) = segments.filter_last_ident("Option") {
+                            let PathArguments::AngleBracketed(generics) = &segment.arguments else {
+                                unreachable!()
                             };
+                            let Some(GenericArgument::Type(Type::Path(this_path))) =
+                                generics.args.last()
+                            else {
+                                return Err(syn::Error::new_spanned(
+                                    generics,
+                                    "where's the generic args?",
+                                ));
+                            };
+                            is_option = true;
+                            segments = this_path.path.segments.clone(); // recurse deeper
+                        } else {
+                            if segments.last_ident_is("SetOfIterator") {
+                                is_setof_iter = true;
+                            } else if segments.last_ident_is("TableIterator") {
+                                is_table_iter = true;
+                            }
+                            break;
                         }
-                        break;
                     }
 
-                    if saw_setof_iterator {
+                    if is_setof_iter {
                         let last_path_segment = option_inner_path.segments.last();
-                        let (used_ty, optional) = match &last_path_segment.map(|ps| &ps.arguments) {
+                        let used_ty = match &last_path_segment.map(|ps| &ps.arguments) {
                             Some(syn::PathArguments::AngleBracketed(args)) => {
                                 match args.args.last().expect("should have one arg?") {
                                     syn::GenericArgument::Type(ty) => {
-                                        match &ty {
-                                            syn::Type::Path(path) => {
-                                                (UsedType::new(syn::Type::Path(path.clone()))?, saw_option_ident)
-                                            }
-                                            syn::Type::Macro(type_macro) => {
-                                                (UsedType::new(syn::Type::Macro(type_macro.clone()), )?, saw_option_ident)
-                                            },
-                                            reference @ syn::Type::Reference(_) => {
-                                                (UsedType::new((*reference).clone(), )?, saw_option_ident)
-                                            },
+                                        match ty {
+                                            Type::Path(_) | Type::Macro(_) | Type::Reference(_) => UsedType::new(ty.clone())?,
```

Comment:
We need to abstract over this "drill into the arguments" pattern, because apparently syn is too pure and holy to ship with such a function.